src/test/java/jenkins/branch/WorkspaceLocatorImplTest.java

```diff
@@ -267,10 +268,10 @@ public void move() throws Exception {
     @Test
     public void collisions() throws Exception {
         WorkspaceLocatorImpl.MODE = WorkspaceLocatorImpl.Mode.ENABLED;
-        FilePath firstWs = r.buildAndAssertSuccess(r.createFreeStyleProject("first-project-with-a-rather-long-name")).getWorkspace();
-        assertEquals("-project-with-a-rather-long-name", firstWs.getName());
+        FilePath firstWs = r.buildAndAssertSuccess(r.createFreeStyleProject("first_project-with-a-rather-long-name")).getWorkspace();
```

Comment:
True, I missed that somehow.

Reply:
I thought it would be better to keep it separated. this tests a collision.
I added a test in the uniquification method. but i'm fine with the change.